src/index/Vocabulary.cpp

```diff
@@ -253,12 +270,21 @@ bool Vocabulary<S, C, I>::getId(const string& word, IndexType* idx) const {
 
 // ___________________________________________________________________________
 template <typename S, typename C, typename I>
-auto Vocabulary<S, C, I>::prefix_range(const string& prefix) const
-    -> std::pair<IndexType, IndexType> {
+auto Vocabulary<S, C, I>::prefix_range_internal_vocabulary(
+    const string& prefix) const -> std::pair<IndexType, IndexType> {
   auto [begin, end] = internalVocabulary_.prefix_range(prefix);
   return {IndexType::make(begin), IndexType::make(end)};
 }
 
+// ___________________________________________________________________________
+template <typename S, typename C, typename I>
+auto Vocabulary<S, C, I>::prefix_range_external_vocabulary(
+    const string& prefix) const -> std::pair<IndexType, IndexType> {
+  auto [begin, end] = externalVocabulary_.prefix_range(prefix);
+  return {IndexType::make(begin + internalVocabulary_.size()),
+          IndexType::make(end + internalVocabulary_.size())};
+}
+
```

Comment:
Check if it would be possible to make the arguments `std::string_view` instead of `const string&`, which is the more modern interface.
If so, consistently apply it on all the levels where you propagate those functions.
(That depends on the implementation of the actual modular vocabularies and on how easy it is to use `string_view` there.

Reply:
(just try it).